source/write/replace.txt

```diff
@@ -0,0 +1,213 @@
+.. _ruby-write-replace:
+
+=================
+Replace Documents
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: modify, change, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to perform a **replace
+operation** on a document in a MongoDB collection. A replace operation
+removes all fields and values from a specified document except the
+``_id`` field, and adds new fields and values that you specify. This
+operation differs from an update operation, which changes only
+specified fields in one or more documents.
+
+To learn more about update operations, see the
+:ref:`ruby-write-update` guide.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your {+language+} application, create a ``MongoClient`` object that connects to an Atlas cluster
+and assign the following values to your ``database`` and ``collection`` variables:
+
+.. literalinclude:: /includes/write/replace.rb
+    :language: ruby
+    :dedent:
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Replace Operation
+-----------------
+
+You can perform a replace operation in MongoDB by using the
+``replace_one`` method. This method removes all fields except the
+``_id`` field from the first document that matches the specified query filter. It
+then adds the fields and values you specify to the empty document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+You must pass the following parameters to the ``replace_one`` method:
+
+- **Query filter**: Matches which documents to update. To learn
+  more about query filters, see the :ref:`ruby-specify-query`
+  guide.
+
+- **Replacement document**: Specifies the fields and values that
+  you want to replace the existing fields and values with.
+
+Replace One Document
+--------------------
```

Comment:
Having this heading here made me think the page was missing a Replace Many header or something. Since this whole page is just about the replace_one method, I don't think this header is necessary

Reply:
Could also move this change this header to be one level down (with `~~~` instead) and call it Replace Example or something similar